server/controllers/shared/updateStatusSelectionController.ts

```diff
@@ -96,13 +96,20 @@ export default class UpdateStatusSelectionController {
         return res.redirect(paths.show.statusHistory({ referralId }))
       }
 
+      const { decisionForCategoryAndReason } = referralStatusUpdateData
+
       if (!reason) {
-        req.flash('reasonError', 'Enter a reason')
+        req.flash(
+          'reasonError',
+          decisionForCategoryAndReason === 'WITHDRAWN'
+            ? 'Enter more information about the reason for withdrawing the referral'
+            : 'Enter a reason for putting the referral on hold',
```

Comment:
This will appear for the other potential status changes, e.g. Not eligible, Suitable not ready, etc